lib/packwerk/cli.rb

```diff
@@ -177,6 +180,33 @@ def check(paths)
       all_offenses.empty?
     end
 
+    def detect_stale_violations(paths)
```

Comment:
There’s no test for this. There are already other subcommands (e.g. `packwerk check`) which have no unit test, but those are covered by the integration tests. Is there an easy way to have _some_ automated test coverage of `packwerk detect-stale-violations`, whether unit or integration, so that it doesn’t get accidentally broken?

Reply:
@rochlefebvre was working on an improvement in this PR https://github.com/Shopify/packwerk/pull/59 so it could make sense to have a look at that.

We are adding more code to the cli class, which makes it harder to maintain. I think it would make sense to move the logic of this command into a separate class, which would make it easier to reason about the code and subsequently test.

Reply:
@doodzik I'm not against moving the logic of it into another class, but I'm not really sure what the class would be. Currently I just wrote an additional test for cli_test.rb as it seemed like that was the best place to put a test for this function as many of the other command line related tests are also there. Anyway, let me know if you think there should be changes. I considered it but also didn't want to break the current pattern.

Reply:
Can you add an issue for this refactor some of the CLI logic out of the CLI so we don't forget?

Reply:
Never mind, I just did it! https://github.com/Shopify/packwerk/issues/70